### DISCUSSION TITLE
Drop Varien_Profiler?

### DISCUSSION THREAD

**Discussion by kyrena** [OpenMage/magento-lts#3538] [Ideas]
It's a question for the core.

**Comment by fballiano** [comment]
I'd agree to drop it, it seems to me something from another age and it has many many calls all over the core

**Comment by Flyingmana** [comment]
It was back then used quite a bit, and most of the Magento 1 specific profiling solutions hook into it.

But with xdebug, as well as modern tracing solutions, there is likely no need for it anymorw

**Comment by kiatng** [comment]
May be for the `next` branch?

**Comment by empiricompany** [comment]
Wait, but after even this extension that I use a lot wouldn't work, right?
https://github.com/madalinoprea/magneto-debug
 It's useful in production when I'm not loading the xdebug extension.
Is there a performance reason? If disabled, I think it does not have an impact on performance.

**Reply by fballiano** [reply]
It would just be to spare probably hundreds of funcion calls per every request.
Also, varien_profile only traces a minor number of methods (that somebody manually coded to trace), but modern solutions (xdebug, blackfire, newrelic) trace every funcion call and every query

**Reply by empiricompany** [reply]
I usually use this extension (which reads varien_profiler) to find bottlenecks quickly in the stack call. 
It's useful for me because it's built-in and I don't have to pay for an external tracer or have xdebug loaded in production. 
IMHO should be kept.

**Reply by fballiano** [reply]
that extension looks nice btw, didn't you fork it and fix the issues yet? ;-)
jokes aside, I never knew about that extension, should I add it to the awesome-openmage or you think it has some bugs that we shouldn't add it?

**Reply by empiricompany** [reply]
> that extension looks nice btw, didn't you fork it and fix the issues yet? ;-)
> jokes aside, I never knew about that extension, should I add it to the awesome-openmage or you think it has some bugs that we shouldn't add it? 

i haven't fork it because I never encountered any issues, but without varien_profiler I don't think it will work

**Reply by fballiano** [reply]
I was asking because the last update is from 2017

**Reply by empiricompany** [reply]
> I was asking because the last update is from 2017

I know but just it works :)

**Reply by fballiano** [reply]
tested it, notes:
- without varien_profiler it won't work
- execution time is always zero
- database query explain triggers an error each time
- seems to work also removing all the calls to varien_profiler::start()/stop() but it won't work without the varien_profiler class itself

so an option could be remove all of this "fake" calls:
<img width="415" alt="Screenshot 2023-09-23 alle 20 20 29" src="https://github.com/OpenMage/magento-lts/assets/909743/a274cdaf-fc7f-415d-90fe-30182430f500">
but keep the varien_profiler class

**Reply by empiricompany** [reply]
we also need to consider that I have seen 3rd party extensions make use of varien_profiler

**Reply by sreichel** [reply]
@fab Fake/Test calls can be removed, but I would keep it for easy debugging. Small shops do not use Blackfire, newrelic and xdebug is often disabled in production.

**Reply by pquerner** [reply]
https://github.com/AOEpeople/Aoe_Profiler is also a thing.

**Comment by pquerner** [comment]
Not sure about removing. I think the way it works right now (for single calls, behind dev mode, whatnot) its only "many" code lines and calls which in production go to the void. Does that add up in the request? Perhaps. But I think (dont have numbers to back this) its very very tiny.

**Reply by Flyingmana** [reply]
likely in the amount of less then a ms, and with modern opcache optimization active, it might not even get actually executed when not activated.

**Comment by addison74** [comment]
From what I can see, it is used in several places, including in this PR #3530. It is possible to be used in extensions that we do not know about. I wouldn't rush to remove it from the core.
